pkg/forwarder/forwarder.go

```diff
@@ -373,6 +374,9 @@ func (f *DefaultForwarder) createPriorityHTTPTransactions(endpoint endpoint, pay
 				t.Headers.Set(apiHTTPHeaderKey, apiKey)
 				t.Headers.Set(versionHTTPHeaderKey, version.AgentVersion)
 				t.Headers.Set(useragentHTTPHeaderKey, fmt.Sprintf("datadog-agent/%s", version.AgentVersion))
+				if config.Datadog.GetBool("allow_arbitrary_tags") {
+					t.Headers.Set(arbitraryTagHTTPHeaderKey, "true")
+				}
```

Comment:
Why not add it all the time and avoid checking the configuration ?

Reply:
Do you mean to default the value to true? If a customer is relying on normalization of their tags in monitors and dashboards (e.g. `user:j@dd.co` -> `user:j_dd.co`), then this will break things for them. It should be an intentional opt-in.

Reply:
I meant always sending the value even if false. But as seen offline we want to leave the default behavior up to the backend.

Reply:
Regarding config check concerns discussed offline, I updated the config check to remove from the nested loop since there is no reason for it to be in there. Here is another config check on the hot path that I think runs for every http request:

https://github.com/DataDog/datadog-agent/blob/b3ff7789eff327a415db078298b4cac2fce47530/pkg/forwarder/transaction.go#L291